spec/controllers/application_controller_spec.rb

```diff
@@ -157,4 +157,75 @@
       expect(assigns(:org_controller)).to eq("vm")
     end
   end
+
+  describe "#build_user_emails_for_edit" do
+    before :each do
+      EvmSpecHelper.local_miq_server
+      MiqUserRole.seed
+
+      role = MiqUserRole.find_by_name("EvmRole-operator")
+
+      group1 = FactoryGirl.create(:miq_group, :miq_user_role => role, :description => "Group1")
+      group2 = FactoryGirl.create(:miq_group, :miq_user_role => role, :description => "Group2")
+      group3 = FactoryGirl.create(:miq_group, :miq_user_role => role, :description => "Group3")
+      current_group = FactoryGirl.create(:miq_group, :miq_user_role => role, :description => "Current Group")
+
+      @user1 = FactoryGirl.create(:user, :userid => "User1", :miq_groups => [group1], :email => "user1@test.com")
+      @user2 = FactoryGirl.create(:user, :userid => "User2", :miq_groups => [group2], :email => "user2@test.com")
+      @user3 = FactoryGirl.create(:user, :userid => "User3", :miq_groups => [group3], :email => "user3@test.com")
+
+      current_users_groups = [current_group, group1, group2]
+      @current_user = FactoryGirl.create(:user, :userid => "Current User", :miq_groups => current_users_groups,
+                                                :email => "current_user@test.com")
+
+      login_as @current_user
+
+      @edit = {:new => {:email => {:to => []}}, :user_emails => []}
+    end
+
+    it "finds users with groups which belongs to current user's groups" do
+      user_ids = User.with_current_user_groups.collect(&:userid)
+      expect(user_ids).to include(@user1.userid)
+      expect(user_ids).to include(@user2.userid)
+      expect(user_ids).not_to include(@user3.userid)
+    end
+
+    it "listing users's emails which belongs to current user's groups" do
+      controller.instance_variable_set(:@edit, @edit)
+
+      expect do
+        controller.send(:build_user_emails_for_edit)
+        @edit = controller.instance_variable_get(:@edit)
+      end.to change { @edit[:user_emails].count }.from(0).to(3)
+
+      @edit = controller.instance_variable_get(:@edit)
+      expect(@edit[:user_emails]).not_to be_nil
+      expect(@edit[:user_emails]).not_to be_empty
+
+      expect(@edit[:user_emails]).to include(@user1.email => "#{@user1.name} (#{@user1.email})")
+      expect(@edit[:user_emails]).to include(@user2.email => "#{@user2.name} (#{@user2.email})")
+      expect(@edit[:user_emails]).to include(@current_user.email => "#{@current_user.name} (#{@current_user.email})")
+      expect(@edit[:user_emails]).not_to include(@user3.email => "#{@user3.name} (#{@user3.email})")
```

Comment:
These assertions seem repetitive, can you not create so many users/groups so we assert one or two things?